test/functional/legacy/quickfix_spec.lua

```diff
@@ -478,6 +528,23 @@ describe('helpgrep', function()
     call('Test_locationlist')
     expected_empty()
   end)
+
+  it('is changed by autocmd', function()
+    call('XquickfixChangedByAutocmd', 'c')
+    expected_empty()
+    call('XquickfixChangedByAutocmd', 'l')
+    expected_empty()
+  end)
+
+  it('does not crash after using caddbuffer with an empty qf list', function()
+    call('Test_caddbuffer_to_empty')
+    expected_empty()
+  end)
+
+  it('cgetexpr does not crash with a NULL element in a list', function()
+    execute('cgetexpr [$x]')
+    eq(2, eval('1+1'))
```

Comment:
I've been tagging these cases with a `-- Still alive?` comment with the intention of extracting it to a e.g. `helpers.is_alive()` function.